processing/src/main/java/io/druid/segment/incremental/OnheapIncrementalIndex.java

```diff
@@ -179,11 +180,11 @@ protected Integer addToFacts(
         aggs[i] = agg.factorize(
             selectors.get(agg.getName())
         );
+        aggs[i].aggregate();
```

Comment:
doAggregate also has proper exception handling for parseExceptions, looks like we should handle them here also.  
can we call doAggregate(...) with the new aggs array here instead ?